fem/nonlininteg.cpp

```diff
@@ -813,4 +816,149 @@ void VectorConvectionNLFIntegrator::AssembleElementGrad(
    }
 }
 
+const IntegrationRule&
+ConvectiveVectorConvectionNLFIntegrator::GetRule(const FiniteElement &fe,
+                                                 ElementTransformation &T)
+{
+   const int order = 2 * fe.GetOrder() + T.OrderGrad(&fe);
+   return IntRules.Get(fe.GetGeomType(), order);
+}
+
+void ConvectiveVectorConvectionNLFIntegrator::AssembleElementGrad(
+   const FiniteElement &el,
+   ElementTransformation &trans,
+   const Vector &elfun,
+   DenseMatrix &elmat)
+{
+   int nd = el.GetDof();
+   int dim = el.GetDim();
+
+   shape.SetSize(nd);
+   dshape.SetSize(nd, dim);
+   dshapex.SetSize(nd, dim);
+   elmat.SetSize(nd * dim);
+   elmat_comp.SetSize(nd);
+   gradEF.SetSize(dim);
+
+   EF.UseExternalData(elfun.GetData(), nd, dim);
+
+   double w;
+   Vector vec1(dim), vec2(dim), vec3(nd);
+
+   const IntegrationRule *ir = IntRule;
+   if (ir == nullptr)
+   {
+      int order = 2 * el.GetOrder() + trans.OrderGrad(&el);
+      ir = &IntRules.Get(el.GetGeomType(), order);
+   }
+
+   elmat = 0.0;
+   for (int i = 0; i < ir->GetNPoints(); i++)
+   {
+      const IntegrationPoint &ip = ir->IntPoint(i);
+      trans.SetIntPoint(&ip);
+
+      el.CalcShape(ip, shape);
+      el.CalcDShape(ip, dshape);
+
+      w = ip.weight;
```

Comment:
Why is it not 
```suggestion
      w = ip.weight * T.Weight();
```
as in `VectorConvectionNLFIntegrator`?

Reply:
I believe you are looking AssembleElementVector in VectorConvectionNLFIntegrator, they do not use that for AssembleElementGrad which is what is being implemented here.

Reply:
My bad, thank you.